drivers/adc/adc_mcux_adc16.c

```diff
@@ -252,19 +326,54 @@ static int mcux_adc16_init(const struct device *dev)
 #endif
 
 	ADC16_Init(base, &adc_config);
-#if defined(FSL_FEATURE_ADC16_HAS_CALIBRATION) && \
-	    FSL_FEATURE_ADC16_HAS_CALIBRATION
+#if defined(FSL_FEATURE_ADC16_HAS_CALIBRATION) &&                              \
+	FSL_FEATURE_ADC16_HAS_CALIBRATION
 	ADC16_SetHardwareAverage(base, kADC16_HardwareAverageCount32);
 	ADC16_DoAutoCalibration(base);
 #endif
 	if (config->channel_mux_b) {
 		ADC16_SetChannelMuxMode(base, kADC16_ChannelMuxB);
 	}
+
+/* Enable HW trigger */
+#ifdef CONFIG_ADC_HW_TRIGGER
+	ADC16_EnableHardwareTrigger(base, true);
+#else
 	ADC16_EnableHardwareTrigger(base, false);
```

Comment:
You could use `IS_ENABLED()` here to improve readability.

Reply:
yes, looks better. Thanks